src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -1240,9 +1240,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           inferExprAlternative(tree, pt)
           adaptAfterOverloadResolution(tree, mode, pt, original)
         case NullaryMethodType(restpe) => // (2)
-          val resTpDeconst =
-            if (isPastTyper || (tree.symbol.isAccessor && tree.symbol.hasFlag(STABLE) && treeInfo.isExprSafeToInline(tree))) restpe
-            else restpe.deconst
+          val resTpDeconst = // only keep the result type as a constant (foldable) type when before erasure & it's safe to inline/fold
```

Comment:
"keep constant types when they are safe to fold. erasure eliminates constant types modulo some exceptions, so keep those."